src/core_functions/scalar/math/numeric.cpp

```diff
@@ -1320,4 +1326,88 @@ ScalarFunctionSet LeastCommonMultipleFun::GetFunctions() {
 	return funcs;
 }
 
+//===--------------------------------------------------------------------===//
+// [divide_by_const]
+//===--------------------------------------------------------------------===//
+
+template <class LHS, class RHS, class RESULT>
+static void ConstRHSDivideOperation(DataChunk &args, ExpressionState &state, Vector &result) {
+	if (args.data[1].GetVectorType() != VectorType::CONSTANT_VECTOR) {
+		throw InvalidInputException("`divide_by_const` called with a non-constant hrs");
+	}
+	RHS rhs = args.data[1].GetValue(0).GetValue<RHS>();
+
+	ValidityMask &validity = FlatVector::Validity(result);
+	if (rhs == 0) {
+		validity.SetInvalid(args.data.size());
```

Comment:
`SetInvalid` sets the given index to null, I guess you want to use `SetAllInvalid`? 
And may be it is better to set result a constant null vector:
```
result.SetVectorType(VectorType::CONSTANT_VECTOR);
ConstantVector::SetNull(result, true);
```

Reply:
Ah I don't know why I didn't notice I wasn't using `SetAllInvalid`. But I changed it to return a constant null vector anyways which is probably better